sdcm/cluster_baremetal.py

```diff
@@ -137,8 +139,13 @@ def _get_node_ips_param(cls, ip_type='public'):
 class MonitorSetPhysical(cluster.BaseMonitorSet, PhysicalMachineCluster):
 
     def __init__(self, **kwargs):
-        self._node_prefix = '%s-%s' % (kwargs.get('user_prefix', cluster.DEFAULT_USER_PREFIX), MONITOR_NAME)
+        user_prefix = kwargs.pop('user_prefix')
+        kwargs.update(dict(
+            cluster_prefix=cluster.prepend_user_prefix(user_prefix, 'monitor-set'),
+            node_prefix=cluster.prepend_user_prefix(user_prefix, 'monitor-node'),
+            node_type='monitor'
+        ))
         cluster.BaseMonitorSet.__init__(self,
                                         targets=kwargs["targets"],
                                         params=kwargs["params"])
-        PhysicalMachineCluster.__init__(self, node_prefix=self._node_prefix, node_type='monitor', **kwargs)
+        PhysicalMachineCluster.__init__(self, **kwargs)
```

Comment:
Why not use a `super`?

Reply:
since this class is inheriting from multiple classes which doesn't have the same signature for `__init__` and need to be called in specific ordering.

it's like that across all the backends right now, not very clean, but to fix need a major refactoring across all backends